.github/actions/update-extension-provider-images/update-images.py

```diff
@@ -0,0 +1,626 @@
+2.  It reads all image entries from the `images.yaml` file.
+3.  It uses the `oci.client` library to query container registries for new tags.
+4.  It updates the `images.yaml` file in-place based on a set of rules:
+    - For patch updates, it updates the tag of the existing entry.
+    - For new minor/major versions, it adds a new image entry.
+    - It ensures the highest minor version has a `targetVersion` of ">= X.Y".
+    - For images without a `targetVersion`, it keeps only the single latest tag.
+5.  It generates a `release-notes.md` file containing direct links to the
+    release pages for all new and intermediate tags.
+
+This script is not intended to be run manually by developers. It should be
+invoked via the `gardener/cc-utils/.github/workflows/update-extension-provider-images.yaml`
+reusable workflow.
+
+Manual Intervention:
+This script handles version *updates*. Manually editing `images.yaml` is still
+required to remove entire image groups for deprecated Kubernetes versions that
+are no longer supported.
+Image Updater Script.
+"""
+
+import argparse
+import yaml
+import re
+import sys
+import semver
+import time
+from typing import Dict, List, Tuple, Optional, TypedDict
+from collections import defaultdict
+from oci import client as oci_client
+from oci.model import OciImageNotFoundException
+from version import parse_to_semver, is_final
+
+
+# --- Type Definitions ---
+class Update(TypedDict):
+    """A structured representation of a single image update."""
+
+    image_name: str
+    old_tag: Optional[str]
+    new_tag: str
+    update_type: str  # 'patch', 'minor', or 'singleton'
+
+
+# --- Helper Functions ---
+def get_tags_from_registry(oci_client: oci_client.Client, repository: str) -> List[str]:
+    """
+    Use the oci.client to retrieve a list of tags for a repository.
+
+    Args:
+        oci_client (oci_client.Client): An initialized OCI client.
+        repository (str): The image repository to query.
+
+    Returns:
+        List[str]: A list of tag strings, e.g., ['v1.2.3', 'v1.2.4'].
+                   Returns an empty list on error.
+    """
+    try:
+        return oci_client.tags(image_reference=repository)
+    except Exception as e:
+        print(f"Error retrieving tags for {repository}: {e}", file=sys.stderr)
+        return []
+
+
+def image_exists(oci_client: oci_client.Client, repository: str, tag: str) -> bool:
+    """
+    Check if a specific image tag exists in the registry by trying to fetch its manifest.
+
+    Args:
+        oci_client (oci_client.Client): An initialized OCI client.
+        repository (str): The image repository.
+        tag (str): The image tag.
+
+    Returns:
+        bool: True if the image exists, False otherwise.
+    """
+    image_ref = f"{repository}:{tag}"
+    retries = 3
+    delay = 2  # seconds
+
+    for i in range(retries):
+        try:
+            oci_client.manifest(image_reference=image_ref, accept="*/*")
```

Comment:
consider using `absent_ok=True`. In this case, absence of requested image will not result in an exception. Also, preferably use `head_manifest` - this does exactly what you need, I think (+ avoids unnecessary transfer of data)

Lastly, you should not have to set the `accept`-header. oci-client's default behaviour should match what you try to do by passing */* (also, some OCI-Registries might not like */*. which is probably not a concern for _this_ script, though).

Reply:
I remove the `image_exists` fctn for now, see [this comment](https://github.com/gardener/cc-utils/pull/1367#discussion_r2362759276)